[RUNTIME] Minor Bug Fix

bodiddles opened the pull request:
Adds a null check to sending param values which was causing some errors. 

Additionally, changes flag for student code state after studentCode terminates to reflect the actual state of studentCode.

bodiddles committed:
[RUNTIME/DAWN] Adds E-stop to Ansible proto

Added E-stop enum/command to the udp proto dawn sends to runtime

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
Merge branch 'master' of https://github.com/pioneers/PieCentral

bodiddles committed:
[RUNTIME] Minor Bug Fixes

Adds a null check for when parsing state for param values which was sometimes crashing code.

Also changes the state of the studentCode status flag in stateManager when studentCode crashes.

nikitakit commented on runtime/testy/Ansible.py:
```diff
@@ -116,7 +116,8 @@ def package(state):
                         param_value_pair = sensor.param_value.add()
                         param_value_pair.param = param
-                        param_value_pair.float_value = value[0]
+                        if value[0]:
```
Do you even want to send the param_value pair if you don't have a value for it?

yizow replied:
I would argue yes. 
It is possible that we do not have any sensors connected (even though its an incorrect configuration because PDB) 

In that scenario, we still want to send the state of the robot via studentCodeState

brandonxxlee replied:
I think Nikita means just for this for loop (the para value pairs), in which case I agree we could probably just skip that one sensor entirely (not skip sending the whole proto)

yizow commented on runtime/testy/runtime.py:
```diff
@@ -113,6 +114,7 @@ def nonTestModePrint(*args):
           break
       stateQueue.put([SM_COMMANDS.RESET, []])
       terminate_process(PROCESS_NAMES.STUDENT_CODE)
+      stateQueue.put([SM_COMMANDS.SET_VAL, [runtime_pb2.RuntimeData.STUDENT_STOPPED, ["studentCodeState"]]])
```
Does this need to be `["studentCodeState"][0]`?

kevinji replied:
Wait, that would just return the string `studentCodeState`...

brandonxxlee reviewed (changes requested):
Should also add `Ansible.py` correctly using the stateManager value of state, not its internally stored value of state.

bodiddles committed:
[RUNTIME] Further bug changes

Changes UDP sender to not send the param or value if the value of the param is None.

Catches connection error with TCP receive and restarts Ansible to mirror expected behavior if connection is lost

Changes studentCode timeout to 1 second to account for the sleep now in studentAPI

yizow commented on runtime/testy/runtime.py:
```diff
@@ -114,7 +114,7 @@ def nonTestModePrint(*args):
       stateQueue.put([SM_COMMANDS.RESET, []])
       terminate_process(PROCESS_NAMES.STUDENT_CODE)
-      stateQueue.put([SM_COMMANDS.SET_VAL, [runtime_pb2.RuntimeData.STUDENT_STOPPED, ["studentCodeState"]]])
+      stateQueue.put([SM_COMMANDS.SET_VAL, [runtime_pb2.RuntimeData.STUDENT_STOPPED, ["studentCodeState"], False]])
```
`set=False` will be more readable i feel. 
Personally i'm a fan of always naming keyword args in function calls, because the fact that theyre optional can be confusing on first read.

yizow committed:
[RUNTIME] explicit naming of keyword argument

brandonxxlee committed:
[RUNTIME] Revert adding default parameter name
